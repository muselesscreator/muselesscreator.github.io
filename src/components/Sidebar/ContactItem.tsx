export type ContactItemProps = {
  IconEl: React.FC;
  title: string;
  href: string;
  value: string;
};
const ContactItem = ({
  IconEl,
  title,
  href,
  value
}: ContactItemProps) => (
  <li className="contact-item">
    <div className="icon-box">
      <IconEl />
    </div>
    <div className="contact-info">
      <p className="contact-title">{title}</p>
      <a href={href} className="contact-link">{value}</a>
    </div>
  </li>
);

export default ContactItem;
