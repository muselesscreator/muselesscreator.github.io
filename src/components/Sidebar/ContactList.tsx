import {
  FaEnvelope,
  FaLinkedin,
  FaGithub,
} from 'react-icons/fa';
import { FaPhone } from 'react-icons/fa6';

import contactInfo from '~/data/contact';

import ContactItem from './ContactItem';

export const fields = [
  {
    IconEl: FaEnvelope,
    title: 'Email',
    value: contactInfo.email,
    href: `mailto:${contactInfo.email}`,
  },
  {
    IconEl: FaPhone,
    title: 'Phone',
    value: contactInfo.phone,
    href: `tel:${contactInfo.phoneRaw}`,
  },
  {
    IconEl: FaGithub,
    title: 'Github',
    value: contactInfo.github,
    href: contactInfo.github,
  },
  {
    IconEl: FaLinkedin,
    title: 'LinkedIn',
    value: contactInfo.linkedIn,
    href: contactInfo.linkedIn,
  },
];
const ContactList = () => (
  <ul className="contacts-list">
    {fields.map((field) => (
      <ContactItem key={field.title} {...field} />
    ))}
  </ul>
);

export default ContactList;
