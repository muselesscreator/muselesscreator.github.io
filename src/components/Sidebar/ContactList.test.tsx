import {
  vi,
  test,
  describe,
  expect,
} from 'vitest';

import { shallow } from '@muselesscreator/react-shallow-snapshot';
import ContactItem from './ContactItem';

import ContactList, { fields } from './ContactList';

vi.mock('./ContactItem', () => ({ default: 'ContactItem' }));
vi.mock('react-icons/fa', () => ({
  FaEnvelope: 'FaEnvelopeIcon',
  FaLinkedin: 'FaLinkedinIcon',
  FaGithub: 'FaGithubIcon',
}));
vi.mock('react-icons/fa6', () => ({
  FaPhone: 'FaPhoneIcon',
}));

describe('ContactList', () => {
  test('renders correctly', () => {
    const el = shallow(<ContactList />);
    expect(el.snapshot).toMatchSnapshot();
  });
  test('contains correct fields for each contact', () => {
    const el = shallow(<ContactList />);
    const items = el.instance.findByType(ContactItem);
    expect(items.length).toBe(fields.length);
    fields.forEach((fieldDef, index) => {
      const field = items[index];
      expect(field.props).toEqual(fieldDef);
    });
  });
});
