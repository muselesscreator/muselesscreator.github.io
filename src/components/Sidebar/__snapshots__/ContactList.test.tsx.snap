// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ContactList > renders correctly 1`] = `
"<ul
  className="contacts-list"
>
  <ContactItem
    IconEl="FaEnvelopeIcon"
    href="mailto:bwarzesk@gmail.com"
    title="Email"
    value="bwarzesk@gmail.com"
  />
  <ContactItem
    IconEl="FaPhoneIcon"
    href="tel:19413875092"
    title="Phone"
    value="+1 (941) 387-5092"
  />
  <ContactItem
    IconEl="FaGithubIcon"
    href="https://www.github.com/muselesscreator"
    title="Github"
    value="https://www.github.com/muselesscreator"
  />
  <ContactItem
    IconEl="FaLinkedinIcon"
    href="https://www.linkedin.com/in/benjamin-warzeski-69111518/"
    title="LinkedIn"
    value="https://www.linkedin.com/in/benjamin-warzeski-69111518/"
  />
</ul>"
`;
